wagtailmenus/templates/menus/bootstrap4/main_menu_dropdown.html

```diff
@@ -0,0 +1,11 @@
+{% load menu_tags %}
+<ul class="navbar-nav mr-auto">
+{% for item in menu_items %}
+    <li class="nav-item {{ item.active_class }}{% if item.has_children_in_menu %} dropdown{% endif %}">
+        <a href="{{ item.href }}" class="nav-link{% if item.has_children_in_menu %} dropdown-toggle" id="ddtoggle_{{ item.link_page.pk }}" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"{% else %}"{% endif %}>{{ item.text }}{% if item.has_children_in_menu %} <span class="caret"></span>{% endif %}</a>
```

Comment:
I think this would be slightly easier to follow if there were 2 if statements on this line - one within `class=""`, and another after it to add the additional attributes.